{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>Allocation of Resources Without Limits or Throttling in GitHub repository ikus060/rdiffweb prior to 2.8.0.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-4138\">https://nvd.nist.gov/vuln/detail/CVE-2023-4138</a></li>\n<li><a href=\"https://github.com/ikus060/rdiffweb/commit/feef0d7b11d86aed29bf98c21526088117964d85\">https://github.com/ikus060/rdiffweb/commit/feef0d7b11d86aed29bf98c21526088117964d85</a></li>\n<li><a href=\"https://huntr.dev/bounties/1b1fa915-d588-4bb1-9e82-6a6be79befed\">https://huntr.dev/bounties/1b1fa915-d588-4bb1-9e82-6a6be79befed</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-wwrg-2w5j-grvx\">https://github.com/advisories/GHSA-wwrg-2w5j-grvx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-wwrg-2w5j-grvx",
            "title": "[rdiffweb] RDiffWeb vulnerable to Allocation of Resources Without Limits or Throttling",
            "date_modified": "2023-09-03T05:01:03.000Z",
            "date_published": "2023-08-03T18:30:35.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-wwrg-2w5j-grvx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Sqlalchemy mako before 1.2.2 is vulnerable to Regular expression Denial of Service when using the Lexer class to parse. This also affects babelplugin and linguaplugin.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-40023\">https://nvd.nist.gov/vuln/detail/CVE-2022-40023</a></li>\n<li><a href=\"https://github.com/sqlalchemy/mako/issues/366\">https://github.com/sqlalchemy/mako/issues/366</a></li>\n<li><a href=\"https://github.com/sqlalchemy/mako/commit/925760291d6efec64fda6e9dd1fd9cfbd5be068c\">https://github.com/sqlalchemy/mako/commit/925760291d6efec64fda6e9dd1fd9cfbd5be068c</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mako/PYSEC-2022-260.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mako/PYSEC-2022-260.yaml</a></li>\n<li><a href=\"https://github.com/sqlalchemy/mako/blob/c2f392e0be52dc67d1b9770ab8cce6a9c736d547/mako/ext/extract.py#L21\">https://github.com/sqlalchemy/mako/blob/c2f392e0be52dc67d1b9770ab8cce6a9c736d547/mako/ext/extract.py#L21</a></li>\n<li><a href=\"https://pyup.io/vulnerabilities/CVE-2022-40023/50870/\">https://pyup.io/vulnerabilities/CVE-2022-40023/50870/</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2022/09/msg00026.html\">https://lists.debian.org/debian-lts-announce/2022/09/msg00026.html</a></li>\n<li><a href=\"https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/\">https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-v973-fxgf-6xhp\">https://github.com/advisories/GHSA-v973-fxgf-6xhp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-v973-fxgf-6xhp",
            "title": "[mako] mako is vulnerable to Regular Expression Denial of Service",
            "date_modified": "2023-09-03T05:00:54.000Z",
            "date_published": "2022-09-16T17:20:25.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-v973-fxgf-6xhp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1. Due to leveraging the Django Template Language's variable resolution logic, the dictsort template filter was potentially vulnerable to information disclosure, or an unintended method call, if passed a suitably crafted key.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-45116\">https://nvd.nist.gov/vuln/detail/CVE-2021-45116</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security/\">https://docs.djangoproject.com/en/4.0/releases/security/</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/jan/04/security-releases/\">https://www.djangoproject.com/weblog/2022/jan/04/security-releases/</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220121-0005/\">https://security.netapp.com/advisory/ntap-20220121-0005/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/</a></li>\n<li><a href=\"https://github.com/django/django/commit/2a8ec7f546d6d5806e221ec948c5146b55bd7489\">https://github.com/django/django/commit/2a8ec7f546d6d5806e221ec948c5146b55bd7489</a></li>\n<li><a href=\"https://github.com/django/django/commit/c7fe895bca06daf12cc1670b56eaf72a1ef27a16\">https://github.com/django/django/commit/c7fe895bca06daf12cc1670b56eaf72a1ef27a16</a></li>\n<li><a href=\"https://github.com/django/django/commit/c9f648ccfac5ab90fb2829a66da4f77e68c7f93a\">https://github.com/django/django/commit/c9f648ccfac5ab90fb2829a66da4f77e68c7f93a</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8c5j-9r9f-c6w8\">https://github.com/advisories/GHSA-8c5j-9r9f-c6w8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8c5j-9r9f-c6w8",
            "title": "[Django] Information disclosure in Django",
            "date_modified": "2023-09-03T05:00:36.000Z",
            "date_published": "2022-01-12T19:21:10.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8c5j-9r9f-c6w8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em>\nWeâ€™d like to disclose an arbitrary code execution vulnerability in <code>jupyter_core</code> that stems from <code>jupyter_core</code> executing untrusted files in the current working directory. This vulnerability allows one user to run code as another.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em>\nUsers should upgrade to <code>jupyter_core&gt;=4.11.2</code>.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em>\nNo</p>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em>\nSimilar advisory in <a href=\"https://github.com/advisories/GHSA-pq7m-3gw7-gq5x\">IPython</a></p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/jupyter/jupyter_core/security/advisories/GHSA-m678-f26j-3hrp\">https://github.com/jupyter/jupyter_core/security/advisories/GHSA-m678-f26j-3hrp</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-39286\">https://nvd.nist.gov/vuln/detail/CVE-2022-39286</a></li>\n<li><a href=\"https://github.com/jupyter/jupyter_core/commit/1118c8ce01800cb689d51f655f5ccef19516e283\">https://github.com/jupyter/jupyter_core/commit/1118c8ce01800cb689d51f655f5ccef19516e283</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/jupyter-core/PYSEC-2022-42974.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/jupyter-core/PYSEC-2022-42974.yaml</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2022/11/msg00022.html\">https://lists.debian.org/debian-lts-announce/2022/11/msg00022.html</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202301-04\">https://security.gentoo.org/glsa/202301-04</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KKMP5OXXIX2QAUNVNJZ5UEQFKDYYJVBA/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KKMP5OXXIX2QAUNVNJZ5UEQFKDYYJVBA/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YIDN7JMLK6AOMBQI4QPSW4MBQGWQ5NIN/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YIDN7JMLK6AOMBQI4QPSW4MBQGWQ5NIN/</a></li>\n<li><a href=\"https://www.debian.org/security/2023/dsa-5422\">https://www.debian.org/security/2023/dsa-5422</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-m678-f26j-3hrp\">https://github.com/advisories/GHSA-m678-f26j-3hrp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-m678-f26j-3hrp",
            "title": "[jupyter-core] Execution with Unnecessary Privileges in JupyterApp",
            "date_modified": "2023-09-02T05:06:18.000Z",
            "date_published": "2022-10-26T22:07:00.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-m678-f26j-3hrp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>mistune.py in Mistune 0.7.4 allows XSS via an unexpected newline (such as in java\\nscript:) or a crafted email address, related to the escape and autolink functions.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2017-15612\">https://nvd.nist.gov/vuln/detail/CVE-2017-15612</a></li>\n<li><a href=\"https://github.com/lepture/mistune/pull/140\">https://github.com/lepture/mistune/pull/140</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-hpv5-v8g5-c864\">https://github.com/advisories/GHSA-hpv5-v8g5-c864</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mistune/PYSEC-2017-80.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mistune/PYSEC-2017-80.yaml</a></li>\n<li><a href=\"https://github.com/lepture/mistune/commit/d6f0b6402299bf5a380e7b4e77bd80e8736630fe\">https://github.com/lepture/mistune/commit/d6f0b6402299bf5a380e7b4e77bd80e8736630fe</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-hpv5-v8g5-c864",
            "title": "[mistune] Cross-site Scripting in Mistune",
            "date_modified": "2023-09-02T05:06:12.000Z",
            "date_published": "2022-05-17T00:26:00.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-hpv5-v8g5-c864"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The (1) FilePathField, (2) GenericIPAddressField, and (3) IPAddressField model field classes in Django before 1.4.11, 1.5.x before 1.5.6, 1.6.x before 1.6.3, and 1.7.x before 1.7 beta 2 do not properly perform type conversion, which allows remote attackers to have unspecified impact and vectors, related to \"MySQL typecasting.\"</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2014-0474\">https://nvd.nist.gov/vuln/detail/CVE-2014-0474</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2014/apr/21/security/\">https://www.djangoproject.com/weblog/2014/apr/21/security/</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-updates/2014-09/msg00023.html\">http://lists.opensuse.org/opensuse-updates/2014-09/msg00023.html</a></li>\n<li><a href=\"http://rhn.redhat.com/errata/RHSA-2014-0456.html\">http://rhn.redhat.com/errata/RHSA-2014-0456.html</a></li>\n<li><a href=\"http://rhn.redhat.com/errata/RHSA-2014-0457.html\">http://rhn.redhat.com/errata/RHSA-2014-0457.html</a></li>\n<li><a href=\"http://www.debian.org/security/2014/dsa-2934\">http://www.debian.org/security/2014/dsa-2934</a></li>\n<li><a href=\"http://www.ubuntu.com/usn/USN-2169-1\">http://www.ubuntu.com/usn/USN-2169-1</a></li>\n<li><a href=\"https://github.com/django/django/commit/5f0829a27e85d89ad8c433f5c6a7a7d17c9e9292\">https://github.com/django/django/commit/5f0829a27e85d89ad8c433f5c6a7a7d17c9e9292</a></li>\n<li><a href=\"https://github.com/django/django/commit/985434fb1d6bf2335bf96c6ebf91c3674f1f399f\">https://github.com/django/django/commit/985434fb1d6bf2335bf96c6ebf91c3674f1f399f</a></li>\n<li><a href=\"https://github.com/django/django/commit/aa80f498de6d687e613860933ac58433ab71ea4b\">https://github.com/django/django/commit/aa80f498de6d687e613860933ac58433ab71ea4b</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-wqjj-hx84-v449\">https://github.com/advisories/GHSA-wqjj-hx84-v449</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-wqjj-hx84-v449",
            "title": "[django] Django Vulnerable to MySQL Injection",
            "date_modified": "2023-09-02T05:05:44.000Z",
            "date_published": "2022-05-17T03:07:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-wqjj-hx84-v449"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>contrib.auth.middleware.RemoteUserMiddleware</code> middleware in Django before 1.4.14, 1.5.x before 1.5.9, 1.6.x before 1.6.6, and 1.7 before release candidate 3, when using the <code>contrib.auth.backends.RemoteUserBackend</code> backend, allows remote authenticated users to hijack web sessions via vectors related to the <code>REMOTE_USER</code> header.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2014-0482\">https://nvd.nist.gov/vuln/detail/CVE-2014-0482</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2014/aug/20/security/\">https://www.djangoproject.com/weblog/2014/aug/20/security/</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-updates/2014-09/msg00023.html\">http://lists.opensuse.org/opensuse-updates/2014-09/msg00023.html</a></li>\n<li><a href=\"http://www.debian.org/security/2014/dsa-3010\">http://www.debian.org/security/2014/dsa-3010</a></li>\n<li><a href=\"https://github.com/django/django/commit/0268b855f9eab3377f2821164ef3e66037789e09\">https://github.com/django/django/commit/0268b855f9eab3377f2821164ef3e66037789e09</a></li>\n<li><a href=\"https://github.com/django/django/commit/5307ce565fbedb9cc27cbe7c757b41a00438d37c\">https://github.com/django/django/commit/5307ce565fbedb9cc27cbe7c757b41a00438d37c</a></li>\n<li><a href=\"https://github.com/django/django/commit/c9e3b9949cd55f090591fbdc4a114fcb8368b6d9\">https://github.com/django/django/commit/c9e3b9949cd55f090591fbdc4a114fcb8368b6d9</a></li>\n<li><a href=\"https://github.com/django/django/commit/dd68f319b365f6cb38c5a6c106faf4f6142d7d88\">https://github.com/django/django/commit/dd68f319b365f6cb38c5a6c106faf4f6142d7d88</a></li>\n<li><a href=\"https://github.com/django/django/blob/aa3cb3f37265be37d892e2b391ff023e9caee2a4/docs/releases/1.5.9.txt#L42\">https://github.com/django/django/blob/aa3cb3f37265be37d892e2b391ff023e9caee2a4/docs/releases/1.5.9.txt#L42</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-625g-gx8c-xcmg\">https://github.com/advisories/GHSA-625g-gx8c-xcmg</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-625g-gx8c-xcmg",
            "title": "[django] Django Middleware Enables Session Hijacking",
            "date_modified": "2023-09-02T05:05:05.000Z",
            "date_published": "2022-05-14T02:09:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-625g-gx8c-xcmg"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Race condition in the <code>xdg.BaseDirectory.get_runtime_dir</code> function in pyxdg 0.25 allows local users to overwrite arbitrary files by pre-creating <code>/tmp/pyxdg-runtime-dir-fallback-victim</code> to point to a victim-owned location, then replacing it with a symlink to an attacker-controlled location once the <code>get_runtime_dir</code> function is called.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2014-1624\">https://nvd.nist.gov/vuln/detail/CVE-2014-1624</a></li>\n<li><a href=\"https://exchange.xforce.ibmcloud.com/vulnerabilities/90618\">https://exchange.xforce.ibmcloud.com/vulnerabilities/90618</a></li>\n<li><a href=\"http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=736247\">http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=736247</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2014/01/21/3\">http://www.openwall.com/lists/oss-security/2014/01/21/3</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2014/01/21/4\">http://www.openwall.com/lists/oss-security/2014/01/21/4</a></li>\n<li><a href=\"https://github.com/takluyver/pyxdg/commit/bd999c1c3fe7ee5f30ede2cf704cf03e400347b4\">https://github.com/takluyver/pyxdg/commit/bd999c1c3fe7ee5f30ede2cf704cf03e400347b4</a></li>\n<li><a href=\"https://web.archive.org/web/20200227194825/http://www.securityfocus.com/bid/65042\">https://web.archive.org/web/20200227194825/http://www.securityfocus.com/bid/65042</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-7372-q459-jxhr\">https://github.com/advisories/GHSA-7372-q459-jxhr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-7372-q459-jxhr",
            "title": "[pyxdg] pyxdg Arbitrary File Overwrite via Race Condition",
            "date_modified": "2023-09-02T05:04:51.000Z",
            "date_published": "2022-05-17T01:27:07.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-7372-q459-jxhr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>When using <code>StaticFiles</code>, if there's a file or directory that starts with the same name as the <code>StaticFiles</code> directory, that file or directory is also exposed via <code>StaticFiles</code> which is a path traversal vulnerability.</p>\n<h3 id=\"details\">Details</h3>\n<p>The root cause of this issue is the usage of <code>os.path.commonprefix()</code>:\n<a href=\"https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174\">https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174</a></p>\n<p>As stated in the Python documentation (<a href=\"https://docs.python.org/3/library/os.path.html#os.path.commonprefix\">https://docs.python.org/3/library/os.path.html#os.path.commonprefix</a>) this function returns the longest prefix common to paths.</p>\n<p>When passing a path like <code>/static/../static1.txt</code>, <code>os.path.commonprefix([full_path, directory])</code> returns <code>./static</code> which is the common part of <code>./static1.txt</code> and <code>./static</code>, It refers to <code>/static/../static1.txt</code> because it is considered in the staticfiles directory. As a result, it becomes possible to view files that should not be open to the public.</p>\n<p>The solution is to use <code>os.path.commonpath</code> as the Python documentation explains that <code>os.path.commonprefix</code> works a character at a time, it does not treat the arguments as paths.</p>\n<h3 id=\"poc\">PoC</h3>\n<p>In order to reproduce the issue, you need to create the following structure:</p>\n<pre><code>â”œâ”€â”€ static\nâ”‚   â”œâ”€â”€ index.html\nâ”œâ”€â”€ static_disallow\nâ”‚   â”œâ”€â”€ index.html\nâ””â”€â”€ static1.txt\n</code></pre>\n<p>And run the <code>Starlette</code> app with:</p>\n<pre><code class=\"language-py\">import uvicorn\nfrom starlette.applications import Starlette\nfrom starlette.routing import Mount\nfrom starlette.staticfiles import StaticFiles\n\n\nroutes = [\n    Mount(\"/static\", app=StaticFiles(directory=\"static\", html=True), name=\"static\"),\n]\n\napp = Starlette(routes=routes)\n\n\nif __name__ == \"__main__\":\n    uvicorn.run(app, host=\"0.0.0.0\", port=8000)\n</code></pre>\n<p>And running the commands:</p>\n<pre><code class=\"language-shell\">curl --path-as-is 'localhost:8000/static/../static_disallow/'\ncurl --path-as-is 'localhost:8000/static/../static1.txt'\n</code></pre>\n<p>The <code>static1.txt</code> and the directory <code>static_disallow</code> are exposed.</p>\n<h3 id=\"impact\">Impact</h3>\n<p>Confidentiality is breached: An attacker may obtain files that should not be open to the public.</p>\n<h3 id=\"credits\">Credits</h3>\n<p>Security researcher <strong>Masashi Yamane of LAC Co., Ltd</strong> reported this vulnerability to <strong>JPCERT/CC Vulnerability Coordination Group</strong> and they contacted us to coordinate a patch for the security issue.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px\">https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px</a></li>\n<li><a href=\"https://github.com/encode/starlette/commit/1797de464124b090f10cf570441e8292936d63e3\">https://github.com/encode/starlette/commit/1797de464124b090f10cf570441e8292936d63e3</a></li>\n<li><a href=\"https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174\">https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/staticfiles.py#L172-L174</a></li>\n<li><a href=\"https://github.com/encode/starlette/releases/tag/0.27.0\">https://github.com/encode/starlette/releases/tag/0.27.0</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-29159\">https://nvd.nist.gov/vuln/detail/CVE-2023-29159</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/starlette/PYSEC-2023-83.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/starlette/PYSEC-2023-83.yaml</a></li>\n<li><a href=\"https://jvn.jp/en/jp/JVN95981715/\">https://jvn.jp/en/jp/JVN95981715/</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-v5gw-mw7f-84px\">https://github.com/advisories/GHSA-v5gw-mw7f-84px</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-v5gw-mw7f-84px",
            "title": "[starlette] Starlette has Path Traversal vulnerability in StaticFiles",
            "date_modified": "2023-09-02T05:04:45.000Z",
            "date_published": "2023-05-17T03:49:14.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-v5gw-mw7f-84px"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>Since Requests v2.3.0, Requests has been vulnerable to potentially leaking <code>Proxy-Authorization</code> headers to destination servers, specifically during redirects to an HTTPS origin. This is a product of how <code>rebuild_proxies</code> is used to recompute and <a href=\"https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/sessions.py#L319-L328\">reattach the <code>Proxy-Authorization</code> header</a> to requests when redirected. Note this behavior has <em>only</em> been observed to affect proxied requests when credentials are supplied in the URL user information component (e.g. <code>https://username:password@proxy:8080</code>).</p>\n<p><strong>Current vulnerable behavior(s):</strong></p>\n<ol>\n<li>HTTP â†’ HTTPS: <strong>leak</strong></li>\n<li>HTTPS â†’ HTTP: <strong>no leak</strong></li>\n<li>HTTPS â†’ HTTPS: <strong>leak</strong></li>\n<li>HTTP â†’ HTTP: <strong>no leak</strong></li>\n</ol>\n<p>For HTTP connections sent through the proxy, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the <code>Proxy-Authorization</code> header must be sent in the CONNECT request as the proxy has no visibility into further tunneled requests. This results in Requests forwarding the header to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate those credentials.</p>\n<p>The reason this currently works for HTTPS connections in Requests is the <code>Proxy-Authorization</code> header is also handled by urllib3 with our usage of the ProxyManager in adapters.py with <a href=\"https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/adapters.py#L199-L235\"><code>proxy_manager_for</code></a>. This will compute the required proxy headers in <code>proxy_headers</code> and pass them to the Proxy Manager, avoiding attaching them directly to the Request object. This will be our preferred option going forward for default usage.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>Starting in Requests v2.31.0, Requests will no longer attach this header to redirects with an HTTPS destination. This should have no negative impacts on the default behavior of the library as the proxy credentials are already properly being handled by urllib3's ProxyManager.</p>\n<p>For users with custom adapters, this <em>may</em> be potentially breaking if you were already working around this behavior. The previous functionality of <code>rebuild_proxies</code> doesn't make sense in any case, so we would encourage any users impacted to migrate any handling of Proxy-Authorization directly into their custom adapter.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>For users who are not able to update Requests immediately, there is one potential workaround.</p>\n<p>You may disable redirects by setting <code>allow_redirects</code> to <code>False</code> on all calls through Requests top-level APIs. Note that if you're currently relying on redirect behaviors, you will need to capture the 3xx response codes and ensure a new request is made to the redirect destination.</p>\n<pre><code>import requests\nr = requests.get('http://github.com/', allow_redirects=False)\n</code></pre>\n<h3 id=\"credits\">Credits</h3>\n<p>This vulnerability was discovered and disclosed by the following individuals.</p>\n<p>Dennis Brinkrolf, Haxolot (<a href=\"https://haxolot.com/\">https://haxolot.com/</a>)\nTobias Funke, (<a href=\"mailto:tobiasfunke93@gmail.com\">tobiasfunke93@gmail.com</a>)</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q\">https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q</a></li>\n<li><a href=\"https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5\">https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-32681\">https://nvd.nist.gov/vuln/detail/CVE-2023-32681</a></li>\n<li><a href=\"https://github.com/psf/requests/releases/tag/v2.31.0\">https://github.com/psf/requests/releases/tag/v2.31.0</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/requests/PYSEC-2023-74.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/requests/PYSEC-2023-74.yaml</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html\">https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-j8r2-6x86-q33q\">https://github.com/advisories/GHSA-j8r2-6x86-q33q</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-j8r2-6x86-q33q",
            "title": "[requests] Unintended leak of Proxy-Authorization header in requests",
            "date_modified": "2023-09-02T05:03:39.000Z",
            "date_published": "2023-05-22T20:36:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-j8r2-6x86-q33q"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>Arbitrary file read when using include file syntax.</p>\n<h3 id=\"details\">Details</h3>\n<p>By using the syntax <code>--8&lt;--\"/etc/passwd\"</code> or <code>--8&lt;--\"/proc/self/environ\"</code> the content of these files will be rendered in the generated documentation. Additionally, a path relative to a specified, allowed base path can also be used to render the content of a file outside the specified base paths: <code>--8&lt;-- \"../../../../etc/passwd\"</code>.</p>\n<p>Within the Snippets extension, there exists a <code>base_path</code> option but the implementation is vulnerable to Directory Traversal.\nThe vulnerable section exists in <code>get_snippet_path(self, path)</code> lines 155 to 174 in snippets.py.</p>\n<pre><code>base = \"docs\"\npath = \"/etc/passwd\"\nfilename = os.path.join(base,path) # Filename is now /etc/passwd\n</code></pre>\n<h3 id=\"poc\">PoC</h3>\n<pre><code class=\"language-py\">import markdown\n\npayload = \"--8&lt;-- \\\"/etc/passwd\\\"\"\nhtml = markdown.markdown(payload, extensions=['pymdownx.snippets'])\n\nprint(html)\n</code></pre>\n<h3 id=\"impact\">Impact</h3>\n<p>Any readable file on the host where the plugin is executing may have its content exposed. This can impact any use of Snippets that exposes the use of Snippets to external users. </p>\n<p>It is never recommended to use Snippets to process user-facing, dynamic content. It is designed to process known content on the backend under the control of the host, but if someone were to accidentally enable it for user-facing content, undesired information could be exposed.</p>\n<h3 id=\"suggestion\">Suggestion</h3>\n<p>Specified snippets should be restricted to the configured, specified base paths as a safe default. Allowing relative or absolute paths that escape the specified base paths would need to be behind a feature switch that must be opt-in and would be at the developer's own risk.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/facelessuser/pymdown-extensions/security/advisories/GHSA-jh85-wwv9-24hv\">https://github.com/facelessuser/pymdown-extensions/security/advisories/GHSA-jh85-wwv9-24hv</a></li>\n<li><a href=\"https://github.com/facelessuser/pymdown-extensions/commit/b7bb4878d6017c03c8dc97c42d8d3bb6ee81db9d\">https://github.com/facelessuser/pymdown-extensions/commit/b7bb4878d6017c03c8dc97c42d8d3bb6ee81db9d</a></li>\n<li><a href=\"https://github.com/facelessuser/pymdown-extensions/releases/tag/10.0\">https://github.com/facelessuser/pymdown-extensions/releases/tag/10.0</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-32309\">https://nvd.nist.gov/vuln/detail/CVE-2023-32309</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jh85-wwv9-24hv\">https://github.com/advisories/GHSA-jh85-wwv9-24hv</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jh85-wwv9-24hv",
            "title": "[pymdown-extensions] Any file can be included with the pymdown-snippets extension",
            "date_modified": "2023-09-02T05:03:24.000Z",
            "date_published": "2023-05-15T20:50:23.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jh85-wwv9-24hv"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>NVFLARE contains a vulnerability in its PKI implementation module, where The CA credentials are transported via pickle and no safe deserialization. The deserialization of Untrusted Data may allow an unprivileged network attacker to cause Remote Code Execution, Denial Of Service, and Impact to both Confidentiality and Integrity.\nAll versions before 2.1.2 are affected.</p>\n<p>CVSS Score = 9.8\n<a href=\"https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fnvd.nist.gov%2Fvuln-metrics%2Fcvss%2Fv3-calculator%3Fvector%3DAV%3AN%2FAC%3AL%2FPR%3AN%2FUI%3AN%2FS%3AU%2FC%3AH%2FI%3AH%2FA%3AH&amp;data=05%7C01%7Cchesterc%40nvidia.com%7Ce9600bde16854b0b380008da4fc544f7%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C637910005925574215%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=5kBrXEmAbqp8R31JCH%2FG95MUly72UPVihnBwiRFmvBY%3D&amp;reserved=0\">AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H</a></p>\n<h3 id=\"patches\">Patches</h3>\n<p>The patch will be included in nvflare==2.1.2</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>Replace pickle serialization with JSON and change the code accordingly</p>\n<p>Additional information\nIssue Found by: Oliver Sellwood (@Nintorac)</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/NVIDIA/NVFlare/security/advisories/GHSA-rcxc-3w2m-mp8h\">https://github.com/NVIDIA/NVFlare/security/advisories/GHSA-rcxc-3w2m-mp8h</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-31604\">https://nvd.nist.gov/vuln/detail/CVE-2022-31604</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/nvflare/PYSEC-2022-231.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/nvflare/PYSEC-2022-231.yaml</a></li>\n<li><a href=\"https://github.com/NVIDIA/NVFlare/commit/fd018eea9dff925a765079a94c2f017920fcda67\">https://github.com/NVIDIA/NVFlare/commit/fd018eea9dff925a765079a94c2f017920fcda67</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rcxc-3w2m-mp8h\">https://github.com/advisories/GHSA-rcxc-3w2m-mp8h</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rcxc-3w2m-mp8h",
            "title": "[nvflare] Unsafe deserialisation in the PKI implementation scheme of NVFlare",
            "date_modified": "2023-09-02T05:03:23.000Z",
            "date_published": "2022-06-22T21:22:42.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rcxc-3w2m-mp8h"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In the python-libnmap package through 0.7.2 for Python, remote command execution can occur (if used in a client application that does not validate arguments).</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-30284\">https://nvd.nist.gov/vuln/detail/CVE-2022-30284</a></li>\n<li><a href=\"https://github.com/savon-noir/python-libnmap/releases\">https://github.com/savon-noir/python-libnmap/releases</a></li>\n<li><a href=\"https://pypi.org/project/python-libnmap/\">https://pypi.org/project/python-libnmap/</a></li>\n<li><a href=\"https://www.swascan.com/security-advisory-libnmap-2/\">https://www.swascan.com/security-advisory-libnmap-2/</a></li>\n<li><a href=\"https://libnmap.readthedocs.io/en/latest/process.html#using-libnmap-process\">https://libnmap.readthedocs.io/en/latest/process.html#using-libnmap-process</a></li>\n<li><a href=\"https://github.com/savon-noir/python-libnmap/commit/c36fecde90017befeb4853396d0e2aac93c95b64\">https://github.com/savon-noir/python-libnmap/commit/c36fecde90017befeb4853396d0e2aac93c95b64</a></li>\n<li><a href=\"https://github.com/savon-noir/python-libnmap/releases/tag/v0.7.3\">https://github.com/savon-noir/python-libnmap/releases/tag/v0.7.3</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-qwqv-j7jr-4hp6\">https://github.com/advisories/GHSA-qwqv-j7jr-4hp6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-qwqv-j7jr-4hp6",
            "title": "[python-libnmap] Argument injection in python-libnmap",
            "date_modified": "2023-09-02T05:03:15.000Z",
            "date_published": "2022-05-06T00:00:54.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-qwqv-j7jr-4hp6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>IPython provides an interactive Python shell and Jupyter kernel to use Python interactively. Versions prior to 8.10.0 are vulnerable to command injection in the <code>set_term_title</code> <a href=\"https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117\">function</a> under specific conditions. This has been patched in version 8.10.0.</p>\n<h3 id=\"impact\">Impact</h3>\n<p>Users are only vulnerable when calling this function in Windows in a Python environment where <a href=\"https://docs.python.org/3/library/ctypes.html\">ctypes</a> is not available. The dependency on ctypes in <code>IPython.utils._process_win32</code> prevents the vulnerable code from ever being reached (making it effectively dead code). However, as a library that could be used by another tool, <code>set_term_title</code> could introduce a vulnerability for dependencies. Currently <code>set_term_title</code> is only called with (semi-)trusted input that contain the current working directory of the current IPython session. If an attacker can control directory names, and manage to get a user to <code>cd</code> into this directory, then the attacker can execute arbitrary commands contained in the folder names.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7\">https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7</a></li>\n<li><a href=\"https://github.com/ipython/ipython/commit/385d69325319a5972ee9b5983638e3617f21cb1f\">https://github.com/ipython/ipython/commit/385d69325319a5972ee9b5983638e3617f21cb1f</a></li>\n<li><a href=\"https://github.com/ipython/ipython/commit/991849c247fc208628879e7ca2923b3c218a5a75\">https://github.com/ipython/ipython/commit/991849c247fc208628879e7ca2923b3c218a5a75</a></li>\n<li><a href=\"https://github.com/Carreau/ipython/blob/7557ade0ed927475d5ab5b573d0ea4febfb22683/docs/source/whatsnew/version8.rst#ipython-810\">https://github.com/Carreau/ipython/blob/7557ade0ed927475d5ab5b573d0ea4febfb22683/docs/source/whatsnew/version8.rst#ipython-810</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-24816\">https://nvd.nist.gov/vuln/detail/CVE-2023-24816</a></li>\n<li><a href=\"https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117\">https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117</a></li>\n<li><a href=\"https://github.com/ipython/ipython/blob/56e6925dfa50e2c7f4a6471547b8176275db7c25/IPython/utils/_process_win32.py#L20\">https://github.com/ipython/ipython/blob/56e6925dfa50e2c7f4a6471547b8176275db7c25/IPython/utils/_process_win32.py#L20</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-29gw-9793-fvw7\">https://github.com/advisories/GHSA-29gw-9793-fvw7</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-29gw-9793-fvw7",
            "title": "[IPython] IPython vulnerable to command injection via set_term_title",
            "date_modified": "2023-09-02T05:03:05.000Z",
            "date_published": "2023-02-10T19:55:53.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-29gw-9793-fvw7"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>ubireader_extract_files is vulnerable to path traversal when run against specifically crafted UBIFS files, allowing the attacker to overwrite files outside of the extraction directory (provided the process has write access to that file or directory). This is due to the fact that a node name (dent_node.name) is considered trusted and joined to the extraction directory path during processing, then the node content is written to that joined path. By crafting a malicious UBIFS file with node names holding path traversal payloads (e.g. ../../tmp/outside.txt), it's possible to force ubi_reader to write outside of the extraction directory. This issue affects ubi-reader before 0.8.5.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-0591\">https://nvd.nist.gov/vuln/detail/CVE-2023-0591</a></li>\n<li><a href=\"https://github.com/jrspruitt/ubi_reader/pull/57\">https://github.com/jrspruitt/ubi_reader/pull/57</a></li>\n<li><a href=\"https://onekey.com/blog/security-advisory-remote-command-execution-in-binwalk/\">https://onekey.com/blog/security-advisory-remote-command-execution-in-binwalk/</a></li>\n<li><a href=\"https://github.com/jrspruitt/ubi_reader/commit/d5d68e6b1b9f7070c29df5f67fc060f579ae9139\">https://github.com/jrspruitt/ubi_reader/commit/d5d68e6b1b9f7070c29df5f67fc060f579ae9139</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vp2x-3mc3-3cj4\">https://github.com/advisories/GHSA-vp2x-3mc3-3cj4</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vp2x-3mc3-3cj4",
            "title": "[ubi-reader] Path traversal in ubi-reader",
            "date_modified": "2023-09-02T05:03:05.000Z",
            "date_published": "2023-01-31T12:30:24.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vp2x-3mc3-3cj4"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Pillow starting with 9.2.0 and prior to 9.3.0 allows denial of service via SAMPLESPERPIXEL. A large value in the SAMPLESPERPIXEL tag could lead to a memory and runtime DOS in TiffImagePlugin.py when setting up the context for image decoding. This issue has been patched in version 9.3.0.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-45199\">https://nvd.nist.gov/vuln/detail/CVE-2022-45199</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/6700\">https://github.com/python-pillow/Pillow/pull/6700</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/2444cddab2f83f28687c7c20871574acbb6dbcf3\">https://github.com/python-pillow/Pillow/commit/2444cddab2f83f28687c7c20871574acbb6dbcf3</a></li>\n<li><a href=\"https://bugs.gentoo.org/878769\">https://bugs.gentoo.org/878769</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/releases/tag/9.3.0\">https://github.com/python-pillow/Pillow/releases/tag/9.3.0</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2022-42980.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2022-42980.yaml</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202211-10\">https://security.gentoo.org/glsa/202211-10</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-q4mp-jvh2-76fj\">https://github.com/advisories/GHSA-q4mp-jvh2-76fj</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-q4mp-jvh2-76fj",
            "title": "[pillow] Pillow subject to DoS via SAMPLESPERPIXEL tag",
            "date_modified": "2023-09-02T05:03:04.000Z",
            "date_published": "2022-11-14T12:00:15.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-q4mp-jvh2-76fj"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Keylime does not enforce that the agent registrar data is the same when the tenant uses it for validation of the EK and identity quote and the verifier for validating the integrity quote. This allows an attacker to use one AK, EK pair from a real TPM to pass EK validation and give the verifier an AK of a software TPM. A successful attack breaks the entire chain of trust because a not validated AK is used by the verifier. This issue is worse if the validation happens first and then the agent gets added to the verifier because the timing is easier and the verifier does not validate the regcount entry being equal to 1. At this time, there are no known workaround.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/keylime/keylime/security/advisories/GHSA-jf66-3q76-h5p5\">https://github.com/keylime/keylime/security/advisories/GHSA-jf66-3q76-h5p5</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-1053\">https://nvd.nist.gov/vuln/detail/CVE-2022-1053</a></li>\n<li><a href=\"https://github.com/keylime/keylime/commit/bd5de712acdd77860e7dc58969181e16c7a8dc5d\">https://github.com/keylime/keylime/commit/bd5de712acdd77860e7dc58969181e16c7a8dc5d</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=2065024,\">https://bugzilla.redhat.com/show_bug.cgi?id=2065024,</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jf66-3q76-h5p5\">https://github.com/advisories/GHSA-jf66-3q76-h5p5</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/keylime/PYSEC-2022-184.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/keylime/PYSEC-2022-184.yaml</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A7WAKVXM7L5D2DUACV6EHA6EJNAX2GVL/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A7WAKVXM7L5D2DUACV6EHA6EJNAX2GVL/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RF6QHU4UGSBATC3HOOE7OP66CYVTR7CV/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RF6QHU4UGSBATC3HOOE7OP66CYVTR7CV/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WEW2PAXO5YGLDLPG45YV2OPLJXJSCECQ/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WEW2PAXO5YGLDLPG45YV2OPLJXJSCECQ/</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jf66-3q76-h5p5",
            "title": "[keylime] Tenant and Verifier might not use the same registrar data",
            "date_modified": "2023-09-02T05:02:56.000Z",
            "date_published": "2022-05-05T15:59:35.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jf66-3q76-h5p5"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An exploitable vulnerability exists in the YAML parsing functionality in the YAMLParser method in Interfaces.py in PyAnyAPI before 0.6.1. A YAML parser can execute arbitrary Python commands resulting in command execution because <code>load</code> is used where <code>safe_load</code> should have been used. An attacker can insert Python into loaded YAML to trigger this vulnerability.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2017-16616\">https://nvd.nist.gov/vuln/detail/CVE-2017-16616</a></li>\n<li><a href=\"https://github.com/Stranger6667/pyanyapi/issues/41\">https://github.com/Stranger6667/pyanyapi/issues/41</a></li>\n<li><a href=\"https://github.com/Stranger6667/pyanyapi/releases/tag/0.6.1\">https://github.com/Stranger6667/pyanyapi/releases/tag/0.6.1</a></li>\n<li><a href=\"https://joel-malwarebenchmark.github.io/blog/2017/11/08/cve-2017-16616-yamlparser-in-pyanyapi/\">https://joel-malwarebenchmark.github.io/blog/2017/11/08/cve-2017-16616-yamlparser-in-pyanyapi/</a></li>\n<li><a href=\"https://pypi.python.org/pypi/pyanyapi/0.6.1\">https://pypi.python.org/pypi/pyanyapi/0.6.1</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vg8g-jpm9-jh8r\">https://github.com/advisories/GHSA-vg8g-jpm9-jh8r</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/pyanyapi/PYSEC-2017-23.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/pyanyapi/PYSEC-2017-23.yaml</a></li>\n<li><a href=\"https://github.com/Stranger6667/pyanyapi/commit/810db626c18ebc261d5f4299d0f0eac38d5eb3cf\">https://github.com/Stranger6667/pyanyapi/commit/810db626c18ebc261d5f4299d0f0eac38d5eb3cf</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vg8g-jpm9-jh8r",
            "title": "[pyanyapi] Unsafe pyyaml load usage in PyAnyAPI",
            "date_modified": "2023-09-02T05:02:55.000Z",
            "date_published": "2022-05-13T01:44:08.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vg8g-jpm9-jh8r"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The cryptography package before 41.0.2 for Python mishandles SSH certificates that have critical options.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-38325\">https://nvd.nist.gov/vuln/detail/CVE-2023-38325</a></li>\n<li><a href=\"https://github.com/pyca/cryptography/issues/9207\">https://github.com/pyca/cryptography/issues/9207</a></li>\n<li><a href=\"https://github.com/pyca/cryptography/pull/9208\">https://github.com/pyca/cryptography/pull/9208</a></li>\n<li><a href=\"https://github.com/pyca/cryptography/compare/41.0.1...41.0.2\">https://github.com/pyca/cryptography/compare/41.0.1...41.0.2</a></li>\n<li><a href=\"https://pypi.org/project/cryptography/#history\">https://pypi.org/project/cryptography/#history</a></li>\n<li><a href=\"https://github.com/pyca/cryptography/commit/1ca7adc97b76a9dfbd3d850628b613eb93b78fc3\">https://github.com/pyca/cryptography/commit/1ca7adc97b76a9dfbd3d850628b613eb93b78fc3</a></li>\n<li><a href=\"https://github.com/pyca/cryptography/pull/7960\">https://github.com/pyca/cryptography/pull/7960</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-112.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-112.yaml</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NMCCTYY3CSNQBFFYYC5DAV6KATHWCUZK/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NMCCTYY3CSNQBFFYYC5DAV6KATHWCUZK/</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230824-0010/\">https://security.netapp.com/advisory/ntap-20230824-0010/</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-cf7p-gm2m-833m\">https://github.com/advisories/GHSA-cf7p-gm2m-833m</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-cf7p-gm2m-833m",
            "title": "[cryptography] cryptography mishandles SSH certificates",
            "date_modified": "2023-09-02T05:02:51.000Z",
            "date_published": "2023-07-14T21:31:08.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-cf7p-gm2m-833m"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>When provided with a URL containing many <code>@</code> characters in the authority component the authority regular expression exhibits catastrophic backtracking causing a denial of service if a URL were passed as a parameter or redirected to via an HTTP redirect.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>The issue has been fixed in urllib3 v1.26.5.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33503\">CVE-2021-33503</a></li>\n<li><a href=\"https://jvn.jp/en/vu/JVNVU92413403/\">JVNVU#92413403 (English)</a></li>\n<li><a href=\"https://jvn.jp/vu/JVNVU92413403/\">JVNVU#92413403 (Japanese)</a></li>\n<li><a href=\"https://github.com/urllib3/urllib3/releases/tag/1.26.5\">urllib3 v1.26.5</a></li>\n</ul>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>If you have any questions or comments about this advisory:</p>\n<ul>\n<li>Ask in our <a href=\"https://discord.gg/urllib3\">community Discord</a></li>\n<li>Email <a href=\"mailto:sethmichaellarson@gmail.com\">sethmichaellarson@gmail.com</a></li>\n</ul>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/urllib3/urllib3/security/advisories/GHSA-q2q7-5pp4-w6pg\">https://github.com/urllib3/urllib3/security/advisories/GHSA-q2q7-5pp4-w6pg</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-33503\">https://nvd.nist.gov/vuln/detail/CVE-2021-33503</a></li>\n<li><a href=\"https://github.com/urllib3/urllib3/commit/2d4a3fee6de2fa45eb82169361918f759269b4ec\">https://github.com/urllib3/urllib3/commit/2d4a3fee6de2fa45eb82169361918f759269b4ec</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-q2q7-5pp4-w6pg\">https://github.com/advisories/GHSA-q2q7-5pp4-w6pg</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FMUGWEAUYGGHTPPXT6YBD53WYXQGVV73/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FMUGWEAUYGGHTPPXT6YBD53WYXQGVV73/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6SCV7ZNAHS3E6PBFLJGENCDRDRWRZZ6W/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6SCV7ZNAHS3E6PBFLJGENCDRDRWRZZ6W/</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202107-36\">https://security.gentoo.org/glsa/202107-36</a></li>\n<li><a href=\"https://www.oracle.com/security-alerts/cpuoct2021.html\">https://www.oracle.com/security-alerts/cpuoct2021.html</a></li>\n<li><a href=\"https://github.com/urllib3/urllib3/commit/5b047b645f5f93900d5e2fc31230848c25eb1f5f#diff-52026d639119bf1e0364836b4e8a18bd9ed3c95c6ba39b26534a5057a65e35bbR65\">https://github.com/urllib3/urllib3/commit/5b047b645f5f93900d5e2fc31230848c25eb1f5f#diff-52026d639119bf1e0364836b4e8a18bd9ed3c95c6ba39b26534a5057a65e35bbR65</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-q2q7-5pp4-w6pg",
            "title": "[urllib3] Catastrophic backtracking in URL authority parser when passed URL containing many @ characters",
            "date_modified": "2023-09-02T05:02:51.000Z",
            "date_published": "2021-06-01T21:19:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-q2q7-5pp4-w6pg"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}